test/llvmpasses/throw-amdgpu.ll

```diff
@@ -0,0 +1,46 @@
+; This file is a part of Julia. License is MIT: https://julialang.org/license
+
+; RUN: opt -load libjulia%shlibext -mtriple=amdgcn-amd-amdhsa -mcpu=gfx900 -AllocOpt -LateLowerGCFrame -S %s | FileCheck %s
```

Comment:
Did you test that this fails on a LLVM without the patches, you might have to call `llc`

Reply:
Why wouldn't `llc` work while `opt` does? Both are built from the same (patched) version of LLVM.

Reply:
Because `llc` actually produces target dependent code.

Reply:
I question the need to have `llc` be able to handle (and emit instructions for) our addrspaces. Is it even defined how a backend should handle some arbitrary addrspace that it doesn't otherwise know anything about? And shouldn't we be coercing our addrspaces into those that the backend is designed to understand?

Reply:
Yes, when we mark the AS as `isNoopAddrSpaceCast` it is supposed to assume `AS 0` IIUC.